lib/libc/Kconfig

```diff
@@ -29,6 +29,15 @@ config MINIMAL_LIBC
 	help
 	  Build with minimal C library.
 
+config PICOLIBC
+	bool "Picolibc library"
+	depends on !NATIVE_APPLICATION
+	select THREAD_LOCAL_STORAGE if ARCH_HAS_THREAD_LOCAL_STORAGE
```

Comment:
Just wondering why TLS is always enabled. Can picolib work without TLS if choose to do so?

Reply:
TLS must be enabled in zephyr if picolibc is built with TLS support. Otherwise, any TLS variables in picolibc will not work. I'm using  'ARCH_HAS_THREAD_LOCAL_STORAGE' as a proxy for whether picolibc was built with TLS support as I don't know how to check for that at configure time (picolibc has settings in picolibc.h which could be used, is there some way to look at those?

Reply:
If picolibc TLS is also always enabled when building for those architectures, this is fine.

Reply:
Well, that depends on the toolchain. ARM ships their GCC embedded toolchain with TLS disabled. As there is no impact on code not using TLS in either case, I assume the ARM people just don't realize they've disabled it. Crosstool-ng and debian both ship with toolchains that enable TLS support in the compiler and picolibc. Once the sdk-ng PR for picolibc is merged, then zephyr can rely on that toolchain for picolibc, where TLS is enabled.

Reply:
In this case, might want to add "&& !(arm toolchain)" for the if condition, since TLS is not required.

Reply:
I think it would be a lot better to figure out if picolibc has TLS support automatically; anyone know enough cmake to do that?